pkg/image/api/helper.go

```diff
@@ -69,23 +70,43 @@ func JoinDockerPullSpec(registry, namespace, name, tag string) string {
 	return fmt.Sprintf("%s/%s/%s%s", registry, namespace, name, tag)
 }
 
-// parseImageName parses image name including a tag and returns image name and tag.
-// copied from kubernetes/pkg/kubelet/dockertools/docker.go#parseImageName
-func parseImageName(image string) (string, string) {
-	tag := ""
-	parts := strings.SplitN(image, "/", 2)
-	repo := ""
-	if len(parts) == 2 {
-		repo = parts[0]
-		image = parts[1]
+// ImageWithMetadata returns a copy of image with the DockerImageMetadata filled in
+// from the raw DockerImageManifest data stored in the image.
+func ImageWithMetadata(image Image) (*Image, error) {
+	if len(image.DockerImageManifest) == 0 {
+		return &image, nil
 	}
-	parts = strings.SplitN(image, ":", 2)
-	if len(parts) == 2 {
-		image = parts[0]
-		tag = parts[1]
+
+	manifestData := image.DockerImageManifest
+
+	image.DockerImageManifest = ""
+
+	manifest := DockerImageManifest{}
+	if err := json.Unmarshal([]byte(image.DockerImageManifest), &manifest); err != nil {
```

Comment:
If you set the `image.DockerImageManifest` to "" on 80, isn't this Unmarshal always unmarshaling the empty string?